### DISCUSSION TITLE
Symmetry of maps in `class_table_inheritance` plugin

### DISCUSSION THREAD

**Discussion by adam12** [jeremyevans/sequel#2193] [General]
(I don't remember raising this before but having deja-vu so maybe I have raised it before)

I've used the `class_table_inheritance` plugin a few times, but this one generally catches me quite frequently.

```ruby
plugin :class_table_inheritance, key: kind,
  model_map: {
    "executive" => :Executive, # Symbol for class
    "staff" => "Staff" # String for class
  },

  table_map: {
    :Executive => "executive_employees", # Symbol for class (this works)
    "Staff" => "staff_employees" # String for class (this silently fails but has symmetry with model_map)
  }
```

When using `String` keys for `table_map`, I've found the queries to be incorrect, with the `INNER JOIN` never being used. I'm presuming the lookup misses.

Is it worth updating the plugin to automatically convert the keys to Symbols in the `table_map` or using a second `fetch` for the String version?

**Comment by jeremyevans** [comment]
`model_map` converts database values (strings by default, but could be other values) into model class names, and both strings and symbols are supported as values (just as `const_get` supports both).  

`table_map` maps class name symbols to subclass table names.  Both keys and values should be symbols, as specified by the documentation.  String values work because Sequel will treat a string table name as an identifier, as you cannot SELECT FROM a string in SQL (that's only due to backwards compatibility, I would drop that behavior otherwise).

Using symbols for `model_map` keys wouldn't work.  So supporting strings for `table_map` keys is not something I would generally be in favor of.  In your example, all strings other than the `model_map` keys should ideally be symbols.

However, Sequel does already have access to the string, since `Class#name` returns a string and not a symbol.  So I guess there is no harm in using a string we already have for lookup, if the lookup by symbol fails.  Can you submit a pull request for that?

**Reply by adam12** [reply]
> Can you submit a pull request for that?

Let me look into this.
